components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -1052,38 +1052,43 @@ mod tests {
         assert!(r.is_none());
     }
 
-    fn make_ctx_about_overflow_truncate_should_clip_to_zero(
+    struct CtxConfig {
         overflow_as_warning: bool,
         truncate_as_warning: bool,
         should_clip_to_zero: bool,
-    ) -> EvalContext {
+        in_insert_stmt: bool,
+        in_update_or_delete_stmt: bool,
+    }
+
+    impl Default for CtxConfig {
+        fn default() -> Self {
+            CtxConfig {
+                overflow_as_warning: false,
+                truncate_as_warning: false,
+                should_clip_to_zero: false,
+                in_insert_stmt: false,
+                in_update_or_delete_stmt: false,
+            }
+        }
+    }
+
+    fn make_ctx(config: CtxConfig) -> EvalContext {
```

Comment:
It's better to `impl From< CtxConfig> for EvalContext` and the following code can be easily `CtxConfig{}.into()`